### PR TITLE
Update Label Maker

### PR DESCRIPTION
**Pull request by jeff-hernandez**
This PR will make the following updates:
- refactor `on_slice` by embedding to `LabelMaker.search`
- rename `time` to `cutoff_time` in the generated label times
- additional test cases on using label maker with mixed offset types
- refactored `transactions` fixture for sharing across testing

### CODE REVIEW

File: composeml/label_maker.py [hunk 1]
```diff
@@ -122,16 +76,39 @@ def search(self, df, minimum_data, num_examples_per_instance, gap, verbose=True,
...
+
+            cutoff_time = offset_time(df.index, minimum_data)
+
+            for example in range(num_examples_per_instance):
+                df = df[cutoff_time:]
+
+                if df.empty:
+                    break
+
+                time = offset_time(df.index, self.window_size)
```
Comment (kmax12) [inline composeml/label_maker.py]: is it possible to give this variable a more descriptive name? perhaps `window_end`

File: composeml/label_maker.py [hunk 2]
```diff
@@ -122,16 +76,39 @@ def search(self, df, minimum_data, num_examples_per_instance, gap, verbose=True,
...
+                label = self.labeling_function(df[:time], *args, **kwargs)
+
+                not_none = label is not None
+                not_nan = label is not pd.np.nan
+                if not_none and not_nan:
```
Comment (kmax12) [inline composeml/label_maker.py]: i believe you can just use `not pd.isnull(label)`

File: composeml/tests/test_label_maker.py [hunk 3]
```diff
@@ -1,75 +1,70 @@
...
+        transactions,
+        minimum_data='1min',
+        num_examples_per_instance=4,
+        gap='3min',
+    )
+
+    pd.testing.assert_frame_equal(given_labels, labels)
+
+
+def test_search_offset_mix_1(transactions, labels):
```
Comment (kmax12) [inline composeml/tests/test_label_maker.py]: in addition to the function name, perhaps add a comment about what mix is testing

e.g
```
Tests
window - relative
minimum_data - absolute
gap - absolute
```

File: composeml/label_maker.py (hunk 1 above)
Comment (jeff-hernandez) [reply composeml/label_maker.py]: Renamed variable in [this](https://github.com/FeatureLabs/compose-ml/pull/26/commits/3faae1725789efb5eca283737381b3151a2ffb5b#diff-22bf41f653d21f0ca3dd15dadf5e01e9R99) commit.

File: composeml/label_maker.py (hunk 2 above)
Comment (jeff-hernandez) [reply composeml/label_maker.py]: Refactor to use `pd.isnull` in [this](https://github.com/FeatureLabs/compose-ml/pull/26/commits/b438280fdd8ad7d75d32a2a44ee1a86ae8e78de6#diff-22bf41f653d21f0ca3dd15dadf5e01e9R102) commit.

File: composeml/tests/test_label_maker.py (hunk 3 above)
Comment (jeff-hernandez) [reply composeml/tests/test_label_maker.py]: Annotated comments on offset tests in [this](https://github.com/FeatureLabs/compose-ml/pull/26/commits/86656326709c219bcd66887016752ea497b42b8a#diff-77d230f17f898fc20eaf609b9d931b91R37) commit.

**Pull request merged**